l10n_fr_siret/models/res_partner.py

```diff
@@ -0,0 +1,167 @@
+                    )
+            if rec.siren:
+                # Check the SIREN type, length and key
+                if not rec.siren.isdigit() or len(rec.siren) != 9:
+                    raise ValidationError(
+                        _(
+                            "The SIREN '{siren}' of partner '{partner_name}' is "
+                            "incorrect: it must have exactly 9 digits."
+                        ).format(siren=rec.siren, partner_name=rec.display_name)
+                    )
+                if not siren_is_valid(rec.siren):
+                    raise ValidationError(
+                        _(
+                            "The SIREN '{siren}' of partner '{partner_name}' is "
+                            "invalid: the checksum is wrong."
+                        ).format(siren=rec.siren, partner_name=rec.display_name)
+                    )
+                # Check the NIC key (you need both SIREN and NIC to check it)
+                if rec.nic and not siret_is_valid(rec.siren + rec.nic):
+                    raise ValidationError(
+                        _(
+                            "The SIRET '{siret}' of partner '{partner_name}' is "
+                            "invalid: the checksum is wrong."
+                        ).format(
+                            siret=(rec.siren + rec.nic), partner_name=rec.display_name
+                        )
+                    )
+
+    @api.model
+    def _commercial_fields(self):
+        # SIREN is the same for the whole company
+        # NIC is different for each address
+        res = super()._commercial_fields()
+        res.append("siren")
+        return res
+
+    @api.model
+    def _address_fields(self):
+        res = super()._address_fields()
+        res.append("nic")
+        return res
+
+    siren = fields.Char(
+        string="SIREN",
+        size=9,
+        tracking=50,
+        help="The SIREN number is the official identity "
+        "number of the company in France. It composes "
+        "the first 9 digits of the SIRET number.",
+    )
+    nic = fields.Char(
+        string="NIC",
+        size=5,
+        tracking=51,
+        help="The NIC number is the official rank number "
+        "of this office in the company in France. It "
+        "composes the last 5 digits of the SIRET "
+        "number.",
+    )
+    # the original SIRET field is definied in l10n_fr
+    # We add an inverse method to make it easier to copy/paste a SIRET
+    # from an external source to the partner form view of Odoo
+    siret = fields.Char(
+        compute="_compute_siret",
+        inverse="_inverse_siret",
+        store=True,
+        readonly=False,
+        help="The SIRET number is the official identity number of this "
+        "company's office in France. It is composed of the 9 digits "
+        "of the SIREN number and the 5 digits of the NIC number, ie. "
+        "14 digits.",
+    )
+    company_registry = fields.Char(
+        size=64,
```

Comment:
Does it makes sense ? I think we could remove that limitation.

Reply:
ok, I removed it